packages/react/src/analytics/integrations/Vitorino/providers/__tests__/Forter.test.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  EnvironmentTypes,
+  EventTypes,
+  SetUserEventData,
+  TrackTypes,
+  TrackTypesValues,
+  utils,
+} from '@farfetch/blackout-analytics';
+import { Forter } from '..';
+import {
+  ForterProductionSiteId,
+  ForterSandboxSiteId,
+  ForterTokenTid,
+} from '../Forter/constants';
+import {
+  loadIntegrationData,
+  trackEventsData,
+} from 'tests/__fixtures__/analytics';
+import { postTracking } from '@farfetch/blackout-client';
+const eventData = trackEventsData[EventTypes.CHECKOUT_STARTED];
+
+let mockPostTrackingsPromiseResolve: (value?: SetUserEventData) => void;
+const mockPostTrackingsPromise = new Promise(
+  resolve => (mockPostTrackingsPromiseResolve = resolve),
+);
+const mockSiteId = 'c3f674e6511e';
+const mockOrigin = 'Whitelabel Portal';
+const mockForterToken =
+  '44807307964b4931a89bf364e580bdc5_1594724271711__UDF43_9ck';
+const mockForterTokenLoadedPromise = Promise.resolve(mockForterToken);
+
+// @ts-expect-error
+const strippedDownAnalytics: StrippedDownAnalytics = {
+  createEvent: (type: TrackTypes) =>
+    Promise.resolve({ ...loadIntegrationData, type }),
+};
+jest.mock('@farfetch/blackout-client', () => {
+  const defaultClient = jest.requireActual('@farfetch/blackout-client');
+
+  return {
+    UserGender: defaultClient.UserGender,
+    postTracking: jest.fn(async () => {
+      mockPostTrackingsPromiseResolve();
+      return await mockPostTrackingsPromise;
+    }),
+  };
+});
+
+jest.mock(
+  '../Forter/ForterTokenLoadedEventDetector',
+  () =>
+    class {
+      getForterToken = jest.fn(async () => await mockForterTokenLoadedPromise);
+    },
+);
+
+utils.logger.error = jest.fn();
+utils.logger.warn = jest.fn();
+
+describe('Forter', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return true on its shouldLoad method', () => {
+    expect(Forter.shouldLoad()).toBe(true);
+  });
+
+  it('should return an instance of Forter class in its createInstance method', () => {
+    expect(
+      Forter.createInstance(
+        { siteId: mockSiteId },
+        loadIntegrationData,
+        strippedDownAnalytics,
+      ),
+    ).toBeInstanceOf(Forter);
+  });
+
+  it('should return default `siteId` if empty value are passed on options', () => {
```

Comment:
```suggestion
  it('should return default `siteId` if an empty value is passed on options', () => {
```
(also valid for test descriptions below)

Reply:
This test was removed.